src/d3a/models/strategy/area_agents/one_sided_engine.py

```diff
@@ -91,12 +97,41 @@ def _delete_forwarded_offer_entries(self, offer):
     def tick(self, *, area):
         self.propagate_offer(area.current_tick)
 
+    def event_offer(self, market_id, offer):
+        if market_id != self.markets.source.id:
+            return
+        if offer.id in self.forwarded_offers:
+            return
+        if offer.id in self.offer_age:
+            return
+        if offer.id not in self.markets.source.offers:
+            # TODO: Potential further optimisation here
+            # This case handles trade events that we receive but are not contained in any of
+            # the source and target market. This happens when another device accepts the offer
+            # before the offer event is dispatched to this market. Current solution is to
```

Comment:
But his only in a split event, right?

Reply:
Indeed, but one can squeeze off some more performance from here. In general, the IAA is a place where it can become way way faster. I just did the obvious fixes, but did not dare to go further after one day of debugging to fix all integration tests.